share/availability/light/availability.go

```diff
@@ -169,6 +175,62 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, header *header
 	return nil
 }
 
+// Prune deletes samples and all sampling data corresponding to provided header from store.
+// The operation will remove all data that ShareAvailable might have created
+func (la *ShareAvailability) Prune(ctx context.Context, h *header.ExtendedHeader) error {
+	dah := h.DAH
+	if share.DataHash(dah.Hash()).IsEmptyEDS() {
+		return nil
+	}
+
+	// Prevent multiple sampling and pruning sessions for the same header height
+	release, err := la.activeHeights.StartSession(ctx, h.Height())
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	key := datastoreKeyForRoot(dah)
+	la.dsLk.RLock()
+	data, err := la.ds.Get(ctx, key)
+	la.dsLk.RUnlock()
+	if errors.Is(err, datastore.ErrNotFound) {
+		// nothing to prune
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("get sampling result: %w", err)
+	}
+
+	var result SamplingResult
+	err = json.Unmarshal(data, &result)
+	if err != nil {
+		return fmt.Errorf("unmarshal sampling result: %w", err)
+	}
+
+	// delete stored samples
+	for _, sample := range result.Available {
```

Comment:
I meant logging if `result.Remaining` is not empty, but the one you made is useful as well